build/lib/src/asset/id.dart

```diff
@@ -72,7 +72,7 @@ class AssetId implements Comparable<AssetId> {
           'An AssetId "from" must be specified to resolve a relative URI');
     }
     return new AssetId(
-        p.normalize(from.package), p.join(p.dirname(from.path), uri));
+        p.normalize(from.package), p.url.join(p.dirname(from.path), uri));
```

Comment:
`p.normalize` and `p.dirname` also need to be changed to the `.url` equivalents I believe